src/common/components/table/filters/table-filters-utils.tsx

```diff
@@ -60,6 +60,25 @@ export const useDateFilterHandler = (useShallow: boolean = true) => {
   };
```

Comment:
This file seems to have a repeated pattern across all filter handlers, each one follows similar structure: building URLSearchParams, applying changes, and updating the URL (router.replace or shallowRouter.replace)

Would you consider abstracting this into a generic testable utility (e.g. useQueryUpdater) that takes a param mutation function and handles the routing logic? It might help reduce duplication and make it easier to add new filters in the future. Also maybe putting all filter keys/query param into enum for better typing.

Reply:
done